parquet-column/src/main/java/org/apache/parquet/column/Comparators.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    public int compare(float f1, float f2) {
+      return Float.compare(f1, f2);
+    }
+
+    /**
+     * Returns the string representation of the specified value for debugging/logging purposes
+     */
+    public String toString(float f) {
+      return String.format("%.5f", f);
+    }
+  }
+
+  public static class DoubleComparator implements Comparator<Double> {
+    @Override
+    public int compare(Double o1, Double o2) {
+      return compare(o1.doubleValue(), o2.doubleValue());
+    }
+
+    public int compare(double d1, double d2) {
+      return Double.compare(d1, d2);
+    }
+
+    /**
+     * Returns the string representation of the specified value for debugging/logging purposes
+     */
+    public String toString(double d) {
+      return String.format("%.5f", d);
+    }
+  }
+
+  public static class BinaryComparator implements Comparator<Binary> {
+    @Override
+    public int compare(Binary o1, Binary o2) {
+      return o1.compareTo(o2);
+    }
+
+    /**
+     * Returns the string representation of the specified value for debugging/logging purposes
+     */
+    public String toString(Binary binary) {
+      return binary.toStringUsingUTF8();
+    }
+  }
+
+  private static final BooleanComparator BOOLEAN_COMPARATOR = new BooleanComparator();
+  private static final IntComparator INT_COMPARATOR = new IntComparator();
+  private static final LongComparator LONG_COMPARATOR = new LongComparator();
+  private static final FloatComparator FLOAT_COMPARATOR = new FloatComparator();
+  private static final DoubleComparator DOUBLE_COMPARATOR = new DoubleComparator();
+  private static final BinaryComparator BINARY_COMPARATOR = new BinaryComparator();
+
+  /**
+   * Returns the proper {@link Comparator} implementation for the specified primitive and logical types. {@code
+   * logicalType} might be {@code null}. In case of the specification does not allow a logical type for the related
+   * primitive type, {@code logicalType} must be {@code null}.
+   */
+  public static Comparator<?> comparator(PrimitiveType.PrimitiveTypeName type, OriginalType logicalType) {
+    switch (type) {
+      case BOOLEAN:
+        return booleanComparator(logicalType);
+      case INT32:
+        return int32Comparator(logicalType);
+      case INT64:
+        return int64Comparator(logicalType);
+      case FLOAT:
+        return floatComparator(logicalType);
+      case DOUBLE:
+        return doubleComparator(logicalType);
+      case INT96:
+      case FIXED_LEN_BYTE_ARRAY:
+      case BINARY:
+        return binaryComparator(type, logicalType);
+      default:
+        throw new UnknownColumnTypeException(type);
+    }
+  }
+
+  public static BooleanComparator booleanComparator(OriginalType logicalType) {
+    if (logicalType != null)
+      throw new IllegalArgumentException("Invalid logical type for BOOLEAN: " + logicalType);
+    return BOOLEAN_COMPARATOR;
+  }
+
+  public static LongComparator int64Comparator(OriginalType logicalType) {
```

Comment:
I think you should delegate to `Type` to supply comparators. That way, you have a stronger guarantee that the logical type and physical type match because `Types` won't allow you to construct an invalid one.